text/0066-separate-context-propagation.md

```diff
@@ -0,0 +1,557 @@
+and implements the protocol as it is defined in that working group.
+
+While Correlations can be used to prototype other cross-cutting concerns, this 
+mechanism is primarily intended to convey values for the OpenTelemetry 
+observability systems. New concerns with different criteria should be modeled 
+separately, using the same underlying context propagation layer as building 
+blocks.
+
+For backwards compatibility, OpenTracing Baggage is propagated as Correlations 
+when using the OpenTracing bridge.
+
+As every correlation adds additional overhead to every request, correlations 
+should be used with care. The implementation MUST propagate these values. It is 
+entirely on the user to choose when to drop correlations, either by removing 
+them via the API, or by filtering them via a network proxy or service mesh.
+
+**`GetCorrelation(context, key) -> value`**  
+To access the value for a label set by a prior event, the Correlations API 
+provides a function which takes a context and a key as input, and returns a value.
+
+**`SetCorrelation(context, key, value) -> context`**  
+To record the value for a label, the Correlations API provides a function which 
+takes a context, a key, and a value as input, and returns an updated context 
+which contains the new value.
+
+**`RemoveCorrelation(context, key) -> context`**  
+To delete a label, the Correlations API provides a function 
+which takes a context and a key as input, and returns an updated context which 
+no longer contains the selected key-value pair.
+
+**`ClearCorrelations(context) -> context`**  
+To avoid sending any labels to an untrusted process, the Correlation API 
+provides a function to remove all Correlations from a context.
+
+**`GetCorrelationPropagator() -> (HTTP_Extractor, HTTP_Injector)`**  
+To deserialize the previous labels set by prior processes, and to serialize the 
+current total set of labels and send them to the next process, the Correlations 
+API provides a function which returns a Correlation-specific implementation of 
+the `HTTPExtract` and `HTTPInject` functions found in the Propagation API.
+
+# Context Propagation
+
+## Context API
+
+Cross-cutting concerns access data in-process using the same, shared context 
+object. Each concern uses its own namespaced set of keys in the context, 
+containing all of the data for that cross-cutting concern.
+
+**`GetValue(context, key) -> value`**  
+To access the local state of an concern, the Context API provides a function 
+which takes a context and a key as input, and returns a value.
+
+**`SetValue(context, key, value) -> context`**  
+To record the local state of a cross-cutting concern, the Context API provides a 
+function which takes a context, a key, and a value as input, and returns an 
+updated context which contains the new value.
+
+
+### Optional: Automated Context Management
+When possible, the OpenTelemetry context should automatically be associated 
```

Comment:
it may be easier for server apps. For client apps if we want this - we need to define "begin" and "end" of context and how to chose it. For instance, should client app start a global context on app start and derive all contexts for button click off of the global one?

Reply:
I agree. Even for languages where there is an automated way of managing context, there are often cases where manage  context must happen explicitly, in order to track the work being done.